### PR TITLE
[nanopb] update to 0.4.5 and add generator

### PR DESCRIPTION
**Pull request by mathisloge**
**Describe the pull request**

- #### What does your PR fix?  
updates nanopb to version 0.4.5 and adds the generator

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
yes

### CODE REVIEW

**Comment by mathisloge** [comment]
mhm. dont actually understand the build error.
```shell
CMake Error at cmake_install.cmake:49 (file):
  file cannot create directory: /proto.  Maybe need administrative
  privileges.
```

**Comment by JackBoosY** [comment]
In https://github.com/nanopb/nanopb/blob/76d835adfa434fa796d5738e934366089b98e124/CMakeLists.txt#L69-L71:
```cmake
install( FILES generator/proto/_utils.py
         DESTINATION ${PYTHON_INSTDIR}/proto/ )
```

**Comment by JackBoosY** [comment]
The variable should be `PYTHON_INSTDIR`.

**Comment by mathisloge** [comment]
@JackBoosY the variable is set here https://github.com/nanopb/nanopb/blob/76d835adfa434fa796d5738e934366089b98e124/CMakeLists.txt#L45 will execute_process not override it?

**Comment by JackBoosY** [comment]
Reproduced by installing `nanopb[generator]`:
```
CMake Error at ports/nanopb/portfile.cmake:48 (file):
  file INSTALL cannot find
  "/home/work/18241/vcpkg/packages/nanopb_x64-linux/Lib/site-packages":
  No such file or directory.
Call Stack (most recent call first):
  scripts/ports.cmake:139 (include)
```

**Review by JackBoosY — APPROVED** [review]

File: ports/nanopb/portfile.cmake [hunk 1]
```diff
@@ -33,5 +40,18 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
+if(${nanopb_BUILD_GENERATOR})
```
Comment (strega-nil-ms) [inline ports/nanopb/portfile.cmake:43]:
```suggestion
if(nanopb_BUILD_GENERATOR)
```

File: ports/nanopb/portfile.cmake [hunk 2]
```diff
@@ -33,5 +40,18 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
...
+    file(INSTALL "${CURRENT_PACKAGES_DIR}/bin/nanopb_generator.py" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+    if(WIN32)
+        file(INSTALL "${CURRENT_PACKAGES_DIR}/bin/protoc-gen-nanopb.bat" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+        file(INSTALL "${CURRENT_PACKAGES_DIR}/Lib/site-packages/" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}/site-packages")
+    else()
+        file(INSTALL "${CURRENT_PACKAGES_DIR}/bin/protoc-gen-nanopb" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+    endif()
+endif()
+
+if(${nanopb_BUILD_STATIC_LIBS})
```
Comment (strega-nil-ms) [inline ports/nanopb/portfile.cmake:53]:
```suggestion
if(nanopb_BUILD_STATIC_LIBS)
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
minor tweak :)

**Comment by mathisloge** [comment]
Dang. Need to update the versions later today

**Comment by JonLiu1993** [comment]
@mathisloge, please run the command "vcpkg x-add-version nanopb --overwrite-version" and submit the changes again

**Comment by mathisloge** [comment]
I know. But I'm only on my phone. So that will happen later today (in around 10 hours)

**Comment by JonLiu1993** [comment]
> I know. But I'm only on my phone. So that will happen later today (in around 10 hours)

I want to do that for you, but it looks like I don’t have permission to do that

**Comment by mathisloge** [comment]
strange @JonLiu1993 i've normally enabled the edit from maintainer option. But I've added it now :)

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR and sorry for the long delay!

**Pull request merged**